src/test/java/unit/WorkspaceUser.java

```diff
@@ -0,0 +1,191 @@
+        continue;
+      }
+      for (IamRole role : user.roles) {
+        // `terra workspace remove-user --email=$email --role=$role
+        TestCommand.runCommandExpectSuccess(
+            "workspace", "remove-user", "--email=" + user.email, "--role=" + role);
+      }
+    }
+
+    super.setupEachTime();
+  }
+
+  @Test
+  @DisplayName("list users includes the workspace creator")
+  void listIncludesCreator() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra workspace list-users --format=json`
+    List<UFWorkspaceUser> listWorkspaceUsers =
+        TestCommand.runCommandExpectSuccess(
+            new TypeReference<>() {}, "workspace", "list-users", "--format=json");
+
+    // check that the workspace creator is in the list as an owner
+    Optional<UFWorkspaceUser> workspaceUser =
+        listWorkspaceUsers.stream()
+            .filter(user -> user.email.equalsIgnoreCase(workspaceCreator.email))
+            .findAny();
+    assertTrue(workspaceUser.isPresent(), "workspace creator is in users list");
+    assertTrue(workspaceUser.get().roles.contains(IamRole.OWNER), "workspace creator is an owner");
+  }
+
+  @Test
+  @DisplayName("list users reflects adding a user")
+  void listReflectAdd() throws IOException {
+    // login as the workspace creator and select a test user to share the workspace with
+    workspaceCreator.login();
+    TestUsers testUser = TestUsers.chooseTestUserWhoIsNot(workspaceCreator);
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra workspace add-user --email=$email --role=READER --format=json
+    UFWorkspaceUser addUserReader =
+        TestCommand.runCommandExpectSuccess(
+            UFWorkspaceUser.class,
+            "workspace",
+            "add-user",
+            "--email=" + testUser.email,
+            "--role=READER",
+            "--format=json");
+
+    // check that the user has the READER role
+    assertTrue(addUserReader.roles.contains(IamRole.READER), "reader role returned by add-user");
+
+    // `terra workspace list-users --format=json`
+    List<UFWorkspaceUser> listWorkspaceUsers =
+        TestCommand.runCommandExpectSuccess(
+            new TypeReference<>() {}, "workspace", "list-users", "--format=json");
+
+    // check that the user is in the list as a reader
+    Optional<UFWorkspaceUser> workspaceUser =
+        listWorkspaceUsers.stream()
+            .filter(user -> user.email.equalsIgnoreCase(testUser.email))
+            .findAny();
+    assertTrue(workspaceUser.isPresent(), "test user is in users list");
+    assertEquals(1, workspaceUser.get().roles.size(), "test user has one role");
+    assertTrue(workspaceUser.get().roles.contains(IamRole.READER), "test user has reader role");
```

Comment:
consider a helper to cut down boilerplate and improve test readability
`private static UFWorkspaceUser expectListedUserWithRoles(String userEmail, IamRoles... roles);`

Reply:
Added two helper methods.